src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
-public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+import java.util.Arrays;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+public abstract class ArrayList<T> implements List<T> {
```

Comment:
The class `ArrayList` is declared as abstract, but it implements all methods from the `List` interface. Consider removing the `abstract` keyword to allow instantiation of this class.